### ISSUE TITLE
Autoconfigure non-compatible with Spring Boot 2.6.0-M3: NoSuchMethodError in `PubSubConsumerRequestSetterTest` and `PubSubProducerRequestSetterTest`

### ISSUE THREAD

**Issue by zhumin8** [GoogleCloudPlatform/spring-cloud-gcp#658]
**Describe the bug**
Testing with Spring Boot 2.6.0-M3 (#599), `PubSubConsumerRequestSetterTest` and `PubSubProducerRequestSetterTest` breaks with NoSuchMethodErrors. 
example of error:
```
java.lang.NoSuchMethodError: 'org.assertj.core.api.IterableAssert org.assertj.core.api.IterableAssert.containsExactly(java.lang.Object[])'

	at brave.test.propagation.PropagationSetterTest.set128(PropagationSetterTest.java:41)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        ... ...
```
Likely brave dependency version needs upgrade. More investigation needed.

**Comment by elefeint** [comment]
Or we are bringing in an old version of assertJ in our project.

**Comment by zhumin8** [comment]
Temp fix applied with #705
